qiskit_algorithms/optimizers/scipy_optimizer.py

```diff
@@ -116,9 +132,27 @@ def minimize(
         jac: Callable[[POINT], POINT] | None = None,
         bounds: list[tuple[float, float]] | None = None,
     ) -> OptimizerResult:
-        # Remove ignored parameters to suppress the warning of scipy.optimize.minimize
-        if self.is_bounds_ignored:
-            bounds = None
+
+        # Overwrite the previous bounds
+        if bounds is not None:
+            self._bounds = bounds
+
+        # Remove ignored bounds to suppress the warning of scipy.optimize.minimize
+        if self.is_bounds_ignored and self._bounds is not None:
+            warnings.warn(
+                f"Optimizer method {self._method} does not support bounds. Bounds ignored.",
+                QiskitAlgorithmsOptimizersWarning,
+            )
+            self._bounds = None
+        elif not self.is_bounds_ignored and self._bounds is None:
+            warnings.warn(
+                f"Optimizer method {self._method} may require defining bounds. "
+                f"Check the Scipy documentation for more info: "
+                f"https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html",
+                QiskitAlgorithmsOptimizersWarning,
```

Comment:
This is logic really only applies to methods (scipy optimizers) that are known about here. Given method is not one we know about then the first test could fail when it does in fact support bounds - here it just defaults to is_bounds_ignored if its not one in the set we know about that supports bounds. 

In the second check, since we know what these optimizers are, thats the only way to is_bounds_ignored False  (at least the present set) do any of them mandate bounds. None is interpreted as no bounds which can also be specified explicitly as (None, None) tuples.

Reply:
For the first part, would you suggest warning/raising errors if the method is not among the known ones, before going through the bound-supported logic?

Reply:
@woodsp-ibm after thinking about it, I see your point. The logic based on `is_bounds_ignored` is a bit loose in checking against bounds, and perhaps should be delegated to a separate discussion if need be.
To keep this PR in scope with resolving https://github.com/qiskit-community/qiskit-machine-learning/issues/570, I will remove these checks and related tests.